### ISSUE TITLE
inconsistent results by tensor.min on different machines

### ISSUE THREAD

**Issue by attardi** [pytorch/pytorch#46989]
## 🐛 Bug

torch.Tensor.min returns different results for the indices on the same tensor on different machines..
This hinders the replicability of experiments.

## To Reproduce

    import torch
    t = torch.tensor([[41., 16., 26., 49.,  7.,  6.,  0.,  8., 76., 31., 54., 48., 28., 73.,
         27., 21., 13., 23., 20., 64., 45., 58.,  3., 15., 43., 38.,  1., 29.,
         24., 33., 22., 19.],
        [40., 15., 25., 48.,  6.,  5.,  1.,  7., 75., 30., 53., 47., 27., 72.,
         26., 20., 12., 22., 19., 63., 44., 57.,  2., 14., 42., 37.,  0., 28.,
         23., 32., 21., 18.],
        [39., 14., 24., 47.,  5.,  4.,  2.,  6., 74., 29., 52., 46., 26., 71.,
         25., 19., 11., 21., 18., 62., 43., 56.,  1., 13., 41., 36.,  1., 27.,
         22., 31., 20., 17.]])
     print(t.min(dim=-1))

On Machine 1 (see below) this produces:

    torch.return_types.min(
    values=tensor([0., 0., 1.]),
    indices=tensor([ 6, 26, 26]))

while on Machine 2 (see below), it produces:

    torch.return_types.min(
    values=tensor([0., 0., 1.]),
    indices=tensor([ 6, 26, 22]))

The second result is correct according to the [documentation](https://pytorch.org/docs/stable/generated/torch.min.html#torch.min).
Nevertheless users should be warned of this difference in behaviour across releases.

## Expected behavior

Same results on all architectures and versions.

### Machine 1

PyTorch version: 1.5.0
Is debug build: False
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: CentOS Linux 7 (Core) (x86_64)
GCC version: (GCC) 4.8.5 20150623 (Red Hat 4.8.5-39)
Clang version: Could not collect
CMake version: version 2.8.12.2

Python version: 3.6 (64-bit runtime)
Is CUDA available: True
CUDA runtime version: Could not collect (11.1)
GPU models and configuration: 
GPU 0: Tesla T4
GPU 1: Tesla T4
GPU 2: Tesla T4
GPU 3: Tesla T4

Nvidia driver version: 455.32.00
cuDNN version: /usr/lib64/libcudnn.so.7.6.5
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.18.2
[pip3] numpydoc==0.9.2
[pip3] pytorch-pretrained-bert==0.6.2
[pip3] pytorch-transformers==1.1.0
[pip3] torch==1.5.0
[conda] Could not collect

### Machine 2

Collecting environment information...
PyTorch version: 1.6.0a0+b31f58d
Is debug build: False
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: Red Hat Enterprise Linux Server release 7.6 (Maipo) (ppc64le)
GCC version: (GCC) 8.4.0
Clang version: Could not collect (10.2)
CMake version: Could not collect

Python version: 3.8 (64-bit runtime)
Is CUDA available: True
CUDA runtime version: 10.2.89
GPU models and configuration: 
GPU 0: Tesla V100-SXM2-16GB
GPU 1: Tesla V100-SXM2-16GB
GPU 2: Tesla V100-SXM2-16GB
GPU 3: Tesla V100-SXM2-16GB

Nvidia driver version: 440.64.00
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.19.2
[pip3] torch==1.6.0a0+b31f58d
[pip3] torchvision==0.7.0a0+78ed10c
[conda] Could not collect

## Additional context

cc @jlin27 @mruberry @heitorschueroff

**Comment by VitalyFedyunin** [comment]
Dimensional min is not deterministic, but you can try to make it so by setting number of threads to one. However it is not documented and not guaranteed to be supported.

I'm leaving this issue open as I think we need to make docs more clear about it.

**Comment by mruberry** [comment]
The docs linked are for PyTorch 1.7 but you're using PyTorch 1.5. The PyTorch 1.5 docs describe the behavior you're seeing: https://pytorch.org/docs/1.5.0/torch.html?highlight=min#torch.min. If you update to PyTorch 1.7 I believe you'll get the behavior you expect.

**Comment by attardi** [comment]
@mruberry Thanks, I had not noticed that.

**Comment by mruberry** [comment]
Please reopen this issue if you're still seeing unexpected behavior on PyTorch 1.7.
